generate/funcs.go

```diff
@@ -142,7 +142,14 @@ func GenerateFunctions(from map[string][]parser.Builtin, categorize bool) []byte
 				continue
 			}
 			args := fn.Types.String()
-			ret := fn.ReturnType.String()
+			retType := fn.ReturnType(nil)
+			// TODO(bdarnell): ReturnType(nil) returns nil for some functions.
+			// Empirically, these should be "anyelement", but is there a better way
+			// to handle this?
+			ret := "anyelement"
```

Comment:
`ReturnType` is now a type to type function which, given some argument types, computes the return type. For example it can say that a function can be applied to any array and returns the array element type.

Ideally we'd like to report the structure of this function. Since in Go we can't print functions, perhaps we'd want an annotation in the builtin struct which represents this function in text. Nathan, thoughts?